src/main/java/net/dv8tion/jda/internal/entities/GuildImpl.java

```diff
@@ -1153,6 +1153,32 @@ public List<GuildVoiceState> getVoiceStates()
                 .collect(Helpers.toUnmodifiableList());
     }
 
+    @Nonnull
+    @Override
+    public CacheRestAction<GuildVoiceState> retrieveMemberVoiceStateById(long id)
+    {
+        JDAImpl jda = getJDA();
+        return new DeferredRestAction<>(jda, GuildVoiceState.class,
+                () ->
+                {
+                    MemberImpl member = (MemberImpl) getMemberById(id);
+                    return member == null ? null : member.getVoiceState();
```

Comment:
If memory serves, you can have members in the cache even without the intents that would keep it up to date. Same for the voice state.
This would mean that this thing would potentially provide out of date information.

@MinnDevelopment am I remembering this correctly?

Reply:
You can't have the voice state cache enabled without enabling the intent. However, the opposite direction is problematic. When the intent is enabled and the member is removed from cache, the voice state information is also lost, even if the member is cached again later.

I would probably just never fallback on cache here.

Reply:
Okay, so in conclusion this should _not_ be a `CachedRestAction`

Reply:
I changed it to a normal `RestAction`